### ISSUE TITLE
1: Website canvas container is not a direct child of the HTML body tag. 2: Off-Canvas Sidebars are not direct children of the HTML body tag

### ISSUE THREAD

**Issue by jcashby** [JoryHogeveen/off-canvas-sidebars#109]
Hi, I keep getting the issues below:

1: Website canvas container is not a direct child of the HTML body tag.
2: Off-Canvas Sidebars are not direct children of the HTML body tag

I have added the code in the header and the footer as per the below:

<body<?php body_class(); ?>>
<?php do_action('website_before'); ?>

and in the footer as below:

<?php do_action('website_after'); ?>

	<?php wp_footer(); ?>

</body>

the website is a test one for now and this was working previously but for some reason, it no longer allows for the offcanvas sidebar to slide open in mobile view

https://giftexperttest.wpengine.com/birthday-gifts/1st-birthday-gifts/

any help would be greatly appreciated.

James

**Comment by JoryHogeveen** [comment]
Hi @jcashby 

In the link you've send I see an element `bodyclass` so I'm guessing there is a space missing in your header.
Please fix that part first!

Cheers, Jory

**Comment by jcashby** [comment]
Hi Jory,

Thanks for the quick response. I'm not an expert coder unfortunately so not sure what you meant with the space missing in the header. This is where I have placed the code:

<body<?php body_class(); ?>>
<?php do_action('website_before'); ?>

Is there something wrong with this line; <body<?php body_class(); ?>

That's the only place 'body' is on the header.php file

**Comment by JoryHogeveen** [comment]
Hi @jcashby 

I see the issue.
Please add a space after `<body` (before `<?php`).

**Comment by jcashby** [comment]
Hi, I added that in but it didn't seem to have changed anything (unless I added the space in the wrong place?).

![Capture](https://user-images.githubusercontent.com/35278247/140762904-0b01477e-0f8b-4a50-a854-c8c7205690e0.PNG)

**Comment by JoryHogeveen** [comment]
I still see the same issue online, did you update the actual server file?

**Comment by jcashby** [comment]
Hi, yes it updates straight away. This is also my OCS settings
![Capture](https://user-images.githubusercontent.com/35278247/140763418-2af136d5-c5bc-4b88-9872-f32b9933f96e.PNG)

**Comment by jcashby** [comment]
I added an underscore and it's showing there but it didn't resolve the problem.
![Capture](https://user-images.githubusercontent.com/35278247/140765642-0d617862-bf1a-4c64-aef0-9ab710dc164b.PNG)

**Comment by JoryHogeveen** [comment]
I just looked again, and now it's good, with a space! :)

However, I do notice that the footer isn't loaded where it should be loaded.
At this moment it's loaded within the canvas container.

This is probably because the original theme has added `<?php wp_footer(); ?>` within the closing div for the `st-container` element.

Could you share your footer.php code?

**Comment by jcashby** [comment]
That's good :) please see below for the footer


![Capture](https://user-images.githubusercontent.com/35278247/140769504-0d9dbefb-bff7-44a3-a189-830225ca28c8.PNG)

**Comment by jcashby** [comment]
Just tried adding wp_footer in (but I haven't changed anything else in the code yet unless I need too?). I use shopkeeper by themeforest. I think my support is well outdated now unfortunately. 

![Capture](https://user-images.githubusercontent.com/35278247/140770657-ebc833ae-b498-4829-ba97-9b9542de6494.PNG)

**Comment by JoryHogeveen** [comment]
Hi @jcashby 

> I use shopkeeper by themeforest.
Themeforest isn't the theme developer, it's the marketplace. From what I know about Themeforest you should still receive updates once you purchase a product. You should definitely update to the latest version.
Last update was in July this year:
 -https://themeforest.net/item/shopkeeper-ecommerce-wp-theme-for-woocommerce/9553045
- https://www.getbowtied.com/documentation/shopkeeper/getting-started/changelog/

You shouldn't need to change anything in your code if your theme is compatible. Most themes are compatible with only known issues with JavaScript (unrelated to these hooks).

If you take a look at your site's code through the element inspector of your browser you'll see that the sidebar element is within your container like so:

```html
<div id="ocs-site" ...>
    <div id="st-container" ...></div>
    <div id="ocs-mobile-4" ...></div>
    ... other scripts and elements.
</div>
```

This is incorrect, it should be like this:
```html
<div id="ocs-site" ...>
    <div id="st-container" ...></div>
</div>
<div id="ocs-mobile-4" ...></div>
... other scripts and elements.
```

Despite the above I also see JavaScript errors. Slidebars cannot initialize for some reason. It looks like it's an issue with your script minifier.

**Comment by jcashby** [comment]
That's interesting. It did previously work for me on Shopkeeper so I'm not sure what has changed. My account has been locked from Envato as someone hacked my account and paid for some applications on there apparently so I'm just waiting to get that back before I can update the theme. I'm on Version 2.9.28. As soon as i can update my theme, I will. I'll also see what else I can do, maybe it's a third party plugin or some other settings I have?

As you say maybe I have stopped script running with Wp Rocket or something similar but I've just deactivated that plugin and it didn't seem to change anything.

**Comment by jcashby** [comment]
just sent you an email on your website email listed in profile

**Comment by jcashby** [comment]
After a lot of troubleshooting. I reverted back to v0.4 and it worked! So I'm not sure whether other people are also having similar issues but it must be something in the new updates that has a conflict somewhere as the set-up in the header / footer is fine. Maybe it is a particular plugin it conflicts with but I also uninstalled pretty much every plugin first, then went through JS troubleshooting. But either way, I'll keep it at this version, thanks.

**Comment by JoryHogeveen** [comment]
> as the set-up in the header / footer is fine.

Sorry but last time I checked your website the hooks setup was not fine (just checked again). It was incorrect according to the requirements of this plugin. The fact that it works with a previous version doesn't change this.

Keeping it on v0.4 is of course your own decision :) My advice is to find a correct way to implement the correct hooks in your theme (new or existing).

Closing this topic!
